IntelliTectAnalyzer/IntelliTectAnalyzer/AnalyzerBlockMixins.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Reflection;
+
+namespace IntelliTectAnalyzer
+{
+    public static class AnalyzerBlockMixins
+    {
+        private static string GetEnumAttributeValue<TAttribute>(Enum value, Func<TAttribute, string> valueAccessor)
+            where TAttribute : Attribute
+        {
+            FieldInfo fi = value.GetType().GetRuntimeField(value.ToString());
+
+            var attributes = (TAttribute[]) fi?.GetCustomAttributes(typeof(TAttribute), false);
+
+            return attributes?.Length > 0 ? valueAccessor(attributes[0]) : null;
+        }
+
+        public static string GetDescription(this AnalyzerBlock value)
+        {
+            if (value == AnalyzerBlock.None)
+            {
+                throw new ArgumentException("Description Attribute must be specified!");
```

Comment:
Should include argument name. And perhaps a message like "AnalyzerBlock must be specified"